File: src/KnightFrank.Antares.Dal/Model/Company/CompanyContact.cs
namespace KnightFrank.Antares.Dal.Model.Company
{
    using System;

    using KnightFrank.Antares.Dal.Model.Contacts;

    public class CompanyContact : BaseEntity
    {
        public Guid ContactId { get; set; }
        public Contact Contact { get; set; }
        public Guid CompanyId { get; set; }
        public Company Company { get; set; }
    }
}

File: src/wwwroot/app/common/models/business/company/companyContact.ts
/// <reference path="../../../../typings/_all.d.ts" />

module Antares.Common.Models.Business {
    export class CompanyContact implements Dto.ICompanyContact {
        id: string;
        contactId: string = '';
        contact: Contact = new Contact();
        companyId: string = '';
        company: Company = new Company();
        
        constructor(contactCompany?: Dto.ICompanyContact, contact?: Dto.IContact, company?: Dto.ICompany) {
            if (contactCompany) {
                angular.extend(this, contactCompany);
                if (contactCompany.company) this.company = new Company(contactCompany.company);
                if (contactCompany.contact) this.contact = new Contact(contactCompany.contact);
            }
            else {
                this.company = new Company(company);
                this.companyId = this.company.id;
                this.contact = new Contact(contact);
                this.contactId = this.contact.id;
            }
        }
    }
}